src/NLog/Config/LoggingConfigurationParser.cs

```diff
@@ -132,14 +111,30 @@ protected void LoadConfig(ILoggingConfigurationElement nlogConfig, string basePa
             }
         }
 
-        private void SetNLogElementSettings(Dictionary<string, string> dict, out bool? parseMessageTemplates, out string internalLogFile)
+        private void SetNLogElementSettings(ILoggingConfigurationElement nlogConfig)
         {
-            parseMessageTemplates = null;
-            internalLogFile = null;
+            var dict = CreateNLogConfigDictionary(nlogConfig);
+
+            bool? parseMessageTemplates = null;
+            bool internalLoggerEnabled = false;
             foreach (var configItem in dict)
             {
                 switch (configItem.Key.ToUpperInvariant())
```

Comment:
PS, I don;t think this is needed because of the `StringComparer.OrdinalIgnoreCase` for the dictionary

Reply:
It is used for the switch-statement. And also it returns a list, not a dict to promise ordering. Will fix the variable name.

Reply:
Fixed the name